### ISSUE TITLE
Please Remove Multi-Language Site Requirement

### ISSUE THREAD

**Issue by luxlogica** [mzur/kirby-uniform#39]
Right now, Uniform requires the developer to enable multi-language functionality in the site. As far as I can see, there are only 15 or so strings that are used by Uniform, and the plugin is using Kirby's multi-language functions simply to use it as a 'string dispenser'. Unfortunately, enabling multi-language functionality in Kirby is not a trivial thing, as it affects many aspects,of the site's functionality. 

Multi-language functionality requires, for instance, that all content text files be named with language-aware extensions: 'home.txt" has to be renamed "home.en.txt", even if you're using a single language. In a site that uses a single language, and has a lot of existing content, this is impractical. If you turn on multi-language, but leave the old content files without changing their extensions, users the have issues when updating their content via the panel - ie, the panel writes the new content to a file with a language-aware extension, but reads it back from the old non-language aware file - making it look to the user as if the content cannot be updated. In the current 2.1 beta version of Kirby, there are numerous multi-language bugs.

Bugs aside, the way that Kirby handles multi-lingual sites is just one of many possible solutions. Kirby's solution is useful only when the client wants to have pretty much exactly the same structure and content in different languages. I have never had a client that wanted this. My multi-lingual sites all have substantially different structure and content, which is more culturally appropriate, so it has always been easier to setup 2 sites, each one in a single language, than to setup a single bilingual site. 

Uniform is a wonderful solution, which can save us a whole lot of time, but having to turn on multi-lingual features is a deal-breaker for me. :(

Would it be at all possible to consider a solution WITHOUT the multi-language setup and string functions? Would it be possible to simply use, for instance, custom 'config' settings by default, and use multi-lingual files only as an optional setup for multi-lingual sites?

**Comment by mzur** [comment]
I'd rather not go away from Kirby's native language support, however buggy it still might be. We should go there and help fixing it instead of writing our own version for this plugin if you ask me :wink: 

As a workaround, you can load the language file manually without having the global multi-language support activated. Just add a line like this to the top of the page controller where you use Uniform:

```php
include_once kirby()->roots()->index.'/site/languages/en.php'
```

**Comment by chbach** [comment]
:+1:  I don’t think that writing a custom localization function is the solution to this problem. Using the language files solely for the plugins without activating the localization feature might be the best workaround here.

**Comment by luxlogica** [comment]
Is it possible for the plugin to check whether localization is turned on for the site, and then load the files 'manually' - as described above - if it isn't'?

**Comment by mzur** [comment]
It is possible, sure, but as I said, I'd rather stick to the native language features of Kirby and not implement any of it again for this plugin. Going down this path, having the multi-language support disabled should be an *exception* which needs a little workaround to function properly. Adding the one line described above shouldn't be too much of an inconvenience. What do you think?

**Comment by luxlogica** [comment]
What I meant was: perhaps Uniform should check whether the developer has turned on multi-language features in the site - it can do that easily by checking, for instance, whether language config variables have been set. If the language configs have been set, then this is a multi-lingual site, and Uniform should proceed as normal. If it isn't, then it already knows this is a single-language site, and should do the one-liner *'include_once'* you have suggested above.

This would be a reasonably simple check that Uniform could do itself, that would make it fully compatible with single-language sites - without requiring any action from the user.

**Comment by mzur** [comment]
Sorry for being quiet for so long. I now saw reason, mainly because implementing this feature removes the nessicity to manually copy/merge the language files when installing the plugin. When Uniform is able to detect language settings and load the files, it can do it automatically regardless the site being single- or multi-language. Thanks for being so patient :-)

**Comment by luxlogica** [comment]
@mzur, has this been added to the current version? The readme/instal instructions still state that multi-language is a requirement...

**Comment by mzur** [comment]
No, it's on the roadmap for v2.2 but I haven't found time to work on it yet.

**Comment by mzur** [comment]
@luxlogica This is now implemented in the [v2.2 dev branch](https://github.com/mzur/kirby-uniform/tree/v2.2). Check out the readme there for how to configure the language on a single language site. Please let me know if you encouter any errors ;-)
